File: java/src/main/java/gov/doc/ntia/sigmf/examples/algorithm/DigitalFilterExample.java
package gov.doc.ntia.sigmf.examples.algorithm;

import gov.doc.ntia.sigmf.Global;
import gov.doc.ntia.sigmf.MetaDoc;
import gov.doc.ntia.sigmf.examples.Example;
import gov.doc.ntia.sigmf.examples.ExampleUtils;
import gov.doc.ntia.sigmf.ext.algorithm.DigitalFilter;
import gov.doc.ntia.sigmf.ext.algorithm.FilterTypeEnum;

import java.util.ArrayList;

public class DigitalFilterExample implements Example {
  @Override
  public MetaDoc getExample() {
    Global global = ExampleUtils.getGlobal();
    global.setVersion("1.0.0");
    DigitalFilter digitalFilter = new DigitalFilter();
    Double[] firCoefficients = new Double[] {1.0, 4.0, 5.0, 3.2};
    digitalFilter.setFilterType(FilterTypeEnum.FIR);
    digitalFilter.setFeedforwardCoefficients(firCoefficients);
    digitalFilter.setFrequencyCutoff(7500000d);
    digitalFilter.setAttenuationCutoff(-6d);
    ArrayList<DigitalFilter> filters = new ArrayList<DigitalFilter>();
    filters.add(digitalFilter);
    MetaDoc metaDoc = new MetaDoc();
    metaDoc.setGlobal(global);
    return metaDoc;
  }
}

File: C#/sigmf/sigmf/examples/algorithm/DigitalFilterExample.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf.examples.algorithm
{
    public class DigitalFilterExample : Example
    {
        public override MetaDoc getExample() {
            MetaDoc metaDoc = new MetaDoc();
            Globals globals = ExampleUtils.getGlobals(getCurrentTime());
            DigitalFilter digFilter = new DigitalFilter();
            List<double> FIRcoefficients = new List<double> { 1.0, 4.0, 5.0, 3.2 };
            digFilter.FilterType = "FIR";
            digFilter.FirCoefficients = FIRcoefficients;
            digFilter.CutoffFrequency = 7500000d;
            digFilter.FrequencyStopband = 7000000d;
            digFilter.AttenuationStopband = -10d;
            digFilter.RipplePassband = -5d;
            digFilter.CutoffAttenuation = -6d;
            List<Annotation> annotations = new List<Annotation>();
            globals.AntiAliasingFilter = digFilter;
            metaDoc.setGlobal(globals);
            metaDoc.setAnnotations(annotations);
            return metaDoc;
        }
        private DateTime getCurrentTime()
        {
            DateTime st = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
            return st;
        }
    }
}
